cpp/include/gpu_operations.h

```diff
@@ -134,6 +165,83 @@ class GpuOperations {
     }
   }
 
+  /// This function multiplies an input Matrix and a Vector
+  ///
+  /// \param a
+  /// Input Matrix
+  /// \param b
+  /// Input Vector of type T
+  ///
+  /// \return
+  /// This function returns a Vector of type T
+  static Matrix<T> Multiply(const Matrix<T> &a, const Vector<T> &b) {
+    if (a.cols() == b.rows() && !a.isZero()) {
+      // Allocate and transfer memories
+      int m = a.rows();
+      int n = b.cols();
+      int k = a.cols();
+
+      const T * h_a = &a(0);
+      const T * h_x = &b(0);
+      Vector<T> h_y(m);
+
+      T * d_a;
+      T * d_x;
+      T * d_y;
+
+      // Setup GPU memory
+      util_->SetupMem(&d_a, h_a, m * k);
+      util_->SetupMem(&d_x, h_x, k * n);
+      util_->SetupMem(&d_y, nullptr, m, false);
+
+      cublasOperation_t norm = CUBLAS_OP_N;
+
+      T alpha = 1.0;
+      T beta = 0.0;
+      int lda = m;
+      int incx = 1;
+      int incy = 1;
+
+      high_resolution_clock::time_point t1 = high_resolution_clock::now();
+      // Set up and do cublas matrix multiply
+      GpuMatrixVectorMul(util_->GetBlasHandle(), norm, m, k, &alpha,
+                        d_a, lda, d_x, incx, &beta, d_y, incy);
+
+      // Device sync
+      util_->SyncDev();
+
+      high_resolution_clock::time_point t2 = high_resolution_clock::now();
```

Comment:
Same as before, please do timing stuff in unit tests.

Reply:
Fixed.